compendium/ConfigurationAdmin/src/ConfigurationAdminImpl.cpp

```diff
@@ -733,7 +802,10 @@ void ConfigurationAdminImpl::PerformAsync(Functor&& f)
       if (incompleteFutures.empty()) {
         futuresCV.notify_one();
       }
-    }));
+    });
+  auto returnFut = fut.share();
+  incompleteFutures.emplace(id, std::move(fut));
```

Comment:
jeff: check to make sure that you can wait on a future, and wait on its shared_future in different code locations